pennylane/devices/default_mixed.py

```diff
@@ -335,6 +335,50 @@ def _apply_channel(self, kraus, wires):
 
         self._state = qnp.einsum(einsum_indices, kraus, self._state, kraus_dagger)
 
+    def _apply_channel_tensordot(self, kraus, wires):
+        r"""Apply a quantum channel specified by a list of Kraus operators to subsystems of the
+        quantum state. For a unitary gate, there is a single Kraus operator.
+
+        Args:
+            kraus (list[array]): Kraus operators
+            wires (Wires): target wires
+        """
+        channel_wires = self.map_wires(wires)
+        num_ch_wires = len(channel_wires)
+
+        # Shape kraus operators and cast them to complex data type
+        kraus_shape = [2] * (num_ch_wires * 2)
+        kraus = [qnp.cast(qnp.reshape(k, kraus_shape), dtype=self.C_DTYPE) for k in kraus]
+
+        # row indices of the quantum state affected by this operation
+        row_wires_list = channel_wires.tolist()
+        # column indices are shifted by the number of wires
+        col_wires_list = [w + self.num_wires for w in row_wires_list]
+
+        channel_col_ids = list(range(num_ch_wires, 2 * num_ch_wires))
+        axes_left = [channel_col_ids, row_wires_list]
+        # Use column indices instead or rows to incorporate transposition of K^\dagger
+        axes_right = [col_wires_list, channel_col_ids]
+        # Apply the Kraus operators, and sum over all Kraus operators afterwards
+        _state = qnp.sum(
+            qnp.stack(
+                [
+                    qnp.tensordot(qnp.tensordot(k, self._state, axes_left), qnp.conj(k), axes_right)
```

Comment:
Readability is much better than the einsum implementation!

Reply:
I agree :D `einsum` is dense, but sometimes really hard to read, in particular when the constraction string is generated automatically.